src/main/java/com/google/sps/storage/CaptionStorage.java

```diff
@@ -0,0 +1,262 @@
+/*
+ * Implementation for DatabaseImpl class
+ *  Current public functions at a glance:
+ *  test()
+ *  addVideo()
+ *  addClause()
+ *  addClauses()
+ *  addMetadata()
+ *  getAllKeywords()
+ *  getTimesForKeyword()
+ *  videoInDb()
+ * 
+ */
+
+package com.google.sps.storage;
+
+import com.google.appengine.api.datastore.*;
+import java.util.*;
+
+public class CaptionStorage implements CaptionStorageInterface {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    private static final String COLUMN_VIDEO = "video";
+    private static final String COLUMN_METADATA = "metadata";
+    private static final String COLUMN_CAPTION = "caption";
+    private static final String COLUMN_TIMES = "timestamps";
+
+    //================================================================================
+    // Public Interface Functions (see DatabaseInterface.java for official descriptions)
+    //================================================================================    
+
+    // add a video to the database
+    public void addVideo(String videoID, String metadata) throws CaptionStorageException {
```

Comment:
I'm not sure about the Java style guide in terms of commenting the method headers of implemented methods when the method header is included in the interface class, so idk if method header comments are needed here or for any other implemented method in this class?